runtime/j9vm/jvm.c

```diff
@@ -1717,12 +1717,56 @@ JNI_CreateJavaVM_impl(JavaVM **pvm, void **penv, void *vm_args, BOOLEAN isJITSer
 	 * trying to load application native libraries that are linked against
 	 * libraries in /usr/lib we could fail to find those libraries if /usr/lib
 	 * is not on the LIBPATH.
-	 * */
-	addToLibpath("/usr/lib", FALSE);
+	 */
+	{
+		/* github #8504 show tests fail if the libpath is modified
+		 * when creating a new process from Java.  The easy fix is to only
+		 * append to the libpath if /usr/lib isn't already appended.
+		 * Example libpath:
+		 * LIBPATH=/jre/lib/ppc64/j9vm:/jre/lib/ppc64:/jre/lib/ppc64/jli:/jre/../lib/ppc64:/usr/lib
+		 */
+
+		const char *currentLibPath = getenv("LIBPATH");
+		BOOLEAN appendToLibPath = TRUE;
+		if (NULL != currentLibPath) {
+			size_t currentLibPathLength = strlen(currentLibPath);
+			const char *usrLib = ":/usr/lib";
+			const UDATA usrLibPtrLength = LITERAL_STRLEN(":/usr/lib");
+			const char *usrLibPtr = strstr(currentLibPath, usrLib);
```

Comment:
This needs to be in a loop, otherwise it won't handle situations like
`LIBPATH=prefix:/usr/lib64:/usr/lib:suffix`

Reply:
Agreed.  I'll ask for another review when I've finished updating